wom/src/main/scala/wdl4s/wdl/WdlNamespace.scala

```diff
@@ -419,9 +419,16 @@ object WdlNamespace {
     expression.variableReferences collect { case variable if container.resolveVariable(variable.terminal.sourceString).isEmpty => variable.terminal }
 
   private def validateDeclaration(declaration: DeclarationInterface, wdlSyntaxErrorFormatter: WdlSyntaxErrorFormatter): Seq[SyntaxError] = {
+    // Used to filter task declaration nodes whose parents are subworkflow calls. Subworkflows will cause task declarations to appear twice among
+    // the namespace's descendants: once with a parent that is the containing static workflow and again with a parent that is the dynamic workflow
+    // call. Looking for variable references when the parent is a workflow call will fail if the variable references a call output since only
+    // declarations are made children of workflow calls. This variable reference check is also redundant to the check that will happen with the
+    // copy of the declaration that has a static workflow parent.
+    def parentIsNotAWorkflowCall(declaration: DeclarationInterface): Boolean = declaration.parent.collect({ case p: WdlWorkflowCall => p }).isEmpty
+
     val invalidVariableReferences = for {
       expr <- declaration.expression.toSeq
-      variable <- referencesToAbsentValues(declaration, expr)
+      variable <- referencesToAbsentValues(declaration, expr) if parentIsNotAWorkflowCall(declaration)
```

Comment:
Could you do this check on line 344 (before we even call `validateDeclaration`) to avoid wasting the effort of re-validating any statically-validated declarations?

Reply:
Looks like it, I can try.